bread/layout/components/datatable.py

```diff
@@ -109,7 +109,7 @@ def __init__(
         rowvariable: str = "row",
         spacing: str = "default",
         zebra: bool = False,
-        **kwargs: dict,
+        **kwargs: str,
```

Comment:
Why is this ```str```? The real type of ```kwargs``` will be ```dict```.

Reply:
```python
def foo(**kwargs):
    print(type(kwargs))

foo() # output: <class 'dict'>
```

Reply:
At runtime it's a `dict`, but the typehint is certainly not `dict`. If you want I can put `Any` instead of `str`?

Reply:
Yes, it actually has to be `Any`

Reply:
> Okay, a lot of comments, I think some of the issues might also originate from when I wrote annotations the first time.
> 
> Another thing I wonder is: Could it make sense that we implement our own type (see https://docs.python.org/3/library/typing.html#user-defined-generic-types) for the cases where we use `htmlgenerator.Lazy`? I hope we could write `Lazy[str]` or `Lazy[Any]` and similar instead of always skipping or using `Union`.
> 
>     1. We express the expected type of the Lazy expression
> 
>     2. No need to always think the "non-lazy" option.

I like the idea of introducing "shortcuts" for the types. They can get very verbose...
But `Lazy[str]` to me just looks like a Lazy that evaluates to a string and not like `Union[Lazy, str]`. I will need to think about this a bit more. Maybe we can introduce an alias for 'Union[Lazy, str, hg.BaseElement]`. But what would we call that? (I guess you would hate `Renderable` ;) )